classes/meson.oeclass

```diff
@@ -0,0 +1,125 @@
+# build rules for meson
+# Needed for latest versions of gnome packages. 
+
+# remember to install
+# python >= 3.5
+# pip3 install meson
+# ninja-build
+
+inherit pkgconfig c c++
+
+def meson_array(d, var):
+    items = d.get(var).split()
+    return repr(items[0] if len(items) == 1 else items)
+
+# Map our ARCH values to what Meson expects:
+# http://mesonbuild.com/Reference-tables.html#cpu-families
+def meson_cpu_family(d, var):
+    import re
+    arch = d.getVar(var)
+    if arch == 'powerpc':
+        return 'ppc'
+    elif arch == 'powerpc64':
+        return 'ppc64'
+    elif arch == 'mipsel':
+        return 'mips'
+    elif arch == 'mips64el':
+        return 'mips64'
+    elif re.match(r"i[3-6]86", arch):
+        return "x86"
+    else:
+        return arch
+
+def meson_endian(d, prefix):
+    endian = d.getVar(prefix + "_ENDIAN")
+    if endian == 'l':
+        return "little"
+    elif endian == 'b':
+        return "big"
+    else:
+        bb.fatal("Cannot determine endianism for %s" % (prefix))
+
+OE_MESON_CROSSFILE = "${B}/meson.cross"
```

Comment:
Should should add
```
B = "${S}/builddir"
```
to avoid cluttering `${S}`

Reply:
Negative. meson must execute in ${S}  
Specifying ${B}/builddir seems to be the meson way to avoid cluttering!

Reply:
Yes, that's the proper way to do this. Also, please remember to add `do_configure[cleandirs] = "${B}"`. And can we set B = WORKDIR/build instead of making it a subdir of `${S}`? I'm worried some repo down the road might have a non-empty `builddir`, for whatever reason, so it seems better to stay completely out of `${S}`, no?

Reply:
oh, sorry, you already do exactly that. Please ignore me.